sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/filesystem/InputFileBuilder.java

```diff
@@ -105,6 +105,8 @@ DefaultInputFile completeAndComputeMetadata(DefaultInputFile inputFile, InputFil
 
     String lang = langDetection.language(inputFile);
     if (lang == null && !settings.getBoolean(CoreProperties.IMPORT_UNKNOWN_FILES_KEY)) {
+      // Return fast to skip costly metadata computation
+      LOG.debug("'{}' language is not supported by any analyzer. Skipping it.", inputFile.relativePath());
```

Comment:
relativePath() is a getter. Since we are not following the get/set naming convention, the rule can't be "smart" about that